tests/test-a101-pins.js

```diff
@@ -26,18 +26,35 @@ for(var i = 0; i < GPIOPins.length; i++) {
     // IO6 and IO9 are defined but unusable as GPIOs currently
     if (pinName == "IO6" || pinName == "IO9") continue;
 
-    var pin = gpio.open({ pin: pins[pinName] });
-    var pinValue = pin.read();
-    assert(typeof pinValue == "boolean",
+    var pinA = gpio.open({ pin: pins[pinName], direction: "in" });
+    var pinAValue1 = pinA.read();
+    var ApullValue = (pinAValue1) ? " weak pull-up" : " weak pull-down";
+
+    assert(typeof pinAValue1 == "boolean",
+           "Arduino101Pins: " + pinName + ApullValue + " on input");
+
+    pinA.write(false);
+    var pinAValue2 = pinA.read();
+
+    assert(typeof pinAValue2 == "boolean" && pinAValue1 == pinAValue2,
            "Arduino101Pins: " + pinName + " input");
```

Comment:
Can you explain what you're trying to test here? The previous assert you were happy whether it was true or false, just checking that it was a boolean. So are you trying to make sure that value doesn't change, that you weren't allowed to write to the input?

I guess that might be worthwhile... but if pinAValue1 was false you wouldn't really be testing anything by writing 'false' to the pin. In that case you should write true. So maybe you should write !pinAValue1?